backend/transcribee_backend/routers/document.py

```diff
@@ -147,6 +151,45 @@ def get_document(
     return document.as_api_document()
 
 
+@document_router.delete("/{document_id}/")
+def delete_document(
+    token: UserToken = Depends(get_user_token),
+    document: Document = Depends(get_document_from_url),
+    session: Session = Depends(get_session),
+) -> None:
+    paths_to_delete = []
+    media_files = select(DocumentMediaFile).where(
+        DocumentMediaFile.document == document
+    )
+
+    for media_file in session.exec(media_files):
+        paths_to_delete.append(settings.storage_path / media_file.file)
+        session.delete(media_file)
+
+    session.execute(delete(DocumentUpdate).where(DocumentUpdate.document == document))
+
+    task_ids = select(Task.id).where(Task.document == document).subquery()
+
+    session.execute(
+        delete(TaskDependency).where(
+            or_(
+                col(TaskDependency.dependant_on_id).in_(task_ids),
+                col(TaskDependency.dependent_task_id).in_(task_ids),
+            )
+        ),
+        execution_options={"synchronize_session": "fetch"},
+    )
+
+    session.execute(delete(Task).where(Task.document == document))
+
+    session.delete(document)
+    session.commit()
+
+    for path in paths_to_delete:
+        os.remove(path)
```

Comment:
If something goes wrong here (i.e. the file does not exist) we get an exception but the document is still gone.
further files will not be deleted and we dont have a handle to them anymore. this is kinda bad?
maybe we want to delete the document as a very last step and collect errors here instead of directly throwing?

Reply:
Indicating to the user that the document is gone, when in fact it is not fully feels bad

Reply:
Catching errors here is a good idea, but I don't think keeping the document around if the deletion fails is a good idea. This would lead to a VERY inconsistent state. I'd prefer something like a cronjob, that deletes files still lying around